src/sqlancer/tidb/TiDBProvider.java

```diff
@@ -213,4 +217,15 @@ protected void executeMutator(int index, TiDBGlobalState globalState) throws Exc
         globalState.executeStatement(queryMutateTable);
     }
 
+    @Override
+    public boolean addRowsToAllTables(TiDBGlobalState globalState) throws Exception {
+        List<TiDBTable> tablesNoRow = globalState.getSchema().getDatabaseTables().stream()
+                .filter(t -> t.getNrRows(globalState) == 0).collect(Collectors.toList());
+        for (TiDBTable table : tablesNoRow) {
+            SQLQueryAdapter queryAddRows = TiDBInsertGenerator.getQuery(globalState, table);
```

Comment:
Based on my understanding, executing this might result in an `ExpectedError`, in which case no row is successfully inserted. Is this an issue? If so, perhaps we can check the number of rows in the table to decide whether to continue trying? In that case, we also need to pay attention that it's not possible to insert a row (e.g., in case of a `CHECK` constraint that always evaluates to `FALSE`). If it's not an issue, we can go ahead and merge.

Reply:
I think it should be fine as it is hard to make sure tables are not non-empty.
 
We have the same logic for other DBMSs, such as https://github.com/sqlancer/sqlancer/blob/main/src/sqlancer/sqlite3/SQLite3Provider.java#L350

Perhaps we can fix them in the future if needed.

Reply:
Could you open an issue for that so we can track this?